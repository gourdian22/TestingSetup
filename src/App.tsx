import { createBrowserRouter, Outlet, RouterProvider } from "react-router-dom";

import { Header } from "./components/Header";
import { Main } from "./components/Main";

import { HomePage } from "./pages/HomePage";
import { Details } from "./pages/Details";
import { NotFound } from "./pages/NotFound";

export const router = createBrowserRouter([
  {
    path: "/",
    element: <App />,
    children: [
      {
        path: "/",
        element: <HomePage />,
      },
      {
        path: "/country/:name",
        element: <Details />,
      },
      {
        path: "*",
        element: <NotFound />,
      },
    ],
  },
]);

export function App() {
  return (
    <>
      <Header />
      <Main>
        <Outlet />
      </Main>
    </>
  );
}

export default App;
