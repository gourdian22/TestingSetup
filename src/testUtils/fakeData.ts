import { Country } from "../types";

export const serverResponseAllCountries = [
  {
    name: "Canada",
    capital: "Ottawa",
    region: "Americas",
    population: 38005238,
    flags: {
      svg: "https://flagcdn.com/ca.svg",
      png: "https://flagcdn.com/w320/ca.png",
    },
    independent: false,
  },
  {
    name: "Mexico",
    capital: "Mexico City",
    region: "Americas",
    population: 128932753,
    flags: {
      svg: "https://flagcdn.com/mx.svg",
      png: "https://flagcdn.com/w320/mx.png",
    },
    independent: false,
  },
  {
    name: "United States of America",
    capital: "Washington, D.C.",
    region: "Americas",
    population: 329484123,
    flags: {
      svg: "https://flagcdn.com/us.svg",
      png: "https://flagcdn.com/w320/us.png",
    },
    independent: false,
  },
];

export const serverResponseUSA = [
  {
    name: "United States of America",
    topLevelDomain: [".us"],
    alpha2Code: "US",
    alpha3Code: "USA",
    callingCodes: ["1"],
    capital: "Washington, D.C.",
    altSpellings: ["US", "USA", "United States of America"],
    subregion: "Northern America",
    region: "Americas",
    population: 329484123,
    latlng: [38.0, -97.0],
    demonym: "American",
    area: 9629091.0,
    gini: 41.4,
    timezones: [
      "UTC-12:00",
      "UTC-11:00",
      "UTC-10:00",
      "UTC-09:00",
      "UTC-08:00",
      "UTC-07:00",
      "UTC-06:00",
      "UTC-05:00",
      "UTC-04:00",
      "UTC+10:00",
      "UTC+12:00",
    ],
    borders: ["CAN", "MEX"],
    nativeName: "United States",
    numericCode: "840",
    flags: {
      svg: "https://flagcdn.com/us.svg",
      png: "https://flagcdn.com/w320/us.png",
    },
    currencies: [
      {
        code: "USD",
        name: "United States dollar",
        symbol: "$",
      },
    ],
    languages: [
      {
        iso639_1: "en",
        iso639_2: "eng",
        name: "English",
        nativeName: "English",
      },
    ],
    translations: {
      br: "Stadoù-Unanet",
      pt: "Estados Unidos",
      nl: "Verenigde Staten",
      hr: "Sjedinjene Američke Države",
      fa: "ایالات متحده آمریکا",
      de: "Vereinigte Staaten von Amerika",
      es: "Estados Unidos",
      fr: "États-Unis",
      ja: "アメリカ合衆国",
      it: "Stati Uniti D'America",
      hu: "Amerikai Egyesült Államok",
    },
    flag: "https://flagcdn.com/us.svg",
    regionalBlocs: [
      {
        acronym: "NAFTA",
        name: "North American Free Trade Agreement",
        otherNames: [
          "Tratado de Libre Comercio de América del Norte",
          "Accord de Libre-échange Nord-Américain",
        ],
      },
    ],
    cioc: "USA",
    independent: true,
  },
];

export const serverResponseMexico = [
  {
    name: "Mexico",
    topLevelDomain: [".mx"],
    alpha2Code: "MX",
    alpha3Code: "MEX",
    callingCodes: ["52"],
    capital: "Mexico City",
    altSpellings: [
      "MX",
      "Mexicanos",
      "United Mexican States",
      "Estados Unidos Mexicanos",
    ],
    subregion: "North America",
    region: "Americas",
    population: 128932753,
    latlng: [23.0, -102.0],
    demonym: "Mexican",
    area: 1964375.0,
    gini: 45.4,
    timezones: ["UTC-08:00", "UTC-07:00", "UTC-06:00"],
    borders: ["BLZ", "GTM", "USA"],
    nativeName: "México",
    numericCode: "484",
    flags: {
      svg: "https://flagcdn.com/mx.svg",
      png: "https://flagcdn.com/w320/mx.png",
    },
    currencies: [
      {
        code: "MXN",
        name: "Mexican peso",
        symbol: "$",
      },
    ],
    languages: [
      {
        iso639_1: "es",
        iso639_2: "spa",
        name: "Spanish",
        nativeName: "Español",
      },
    ],
    translations: {
      br: "Mec'hiko",
      pt: "México",
      nl: "Mexico",
      hr: "Meksiko",
      fa: "مکزیک",
      de: "Mexiko",
      es: "México",
      fr: "Mexique",
      ja: "メキシコ",
      it: "Messico",
      hu: "Mexikó",
    },
    flag: "https://flagcdn.com/mx.svg",
    regionalBlocs: [
      {
        acronym: "PA",
        name: "Pacific Alliance",
        otherNames: ["Alianza del Pacífico"],
      },
      {
        acronym: "NAFTA",
        name: "North American Free Trade Agreement",
        otherNames: [
          "Tratado de Libre Comercio de América del Norte",
          "Accord de Libre-échange Nord-Américain",
        ],
      },
    ],
    cioc: "MEX",
    independent: true,
  },
];

export const serverResponseUkraine = [
  {
    name: "Ukraine",
    topLevelDomain: [".ua"],
    alpha2Code: "UA",
    alpha3Code: "UKR",
    callingCodes: ["380"],
    capital: "Kyiv",
    altSpellings: ["UA", "Ukrayina"],
    subregion: "Eastern Europe",
    region: "Europe",
    population: 44134693,
    latlng: [49.0, 32.0],
    demonym: "Ukrainian",
    area: 603700.0,
    gini: 26.6,
    timezones: ["UTC+02:00"],
    borders: ["BLR", "HUN", "MDA", "POL", "ROU", "RUS", "SVK"],
    nativeName: "Україна",
    numericCode: "804",
    flags: {
      svg: "https://flagcdn.com/ua.svg",
      png: "https://flagcdn.com/w320/ua.png",
    },
    currencies: [
      {
        code: "UAH",
        name: "Ukrainian hryvnia",
        symbol: "₴",
      },
    ],
    languages: [
      {
        iso639_1: "uk",
        iso639_2: "ukr",
        name: "Ukrainian",
        nativeName: "Українська",
      },
    ],
    translations: {
      br: "Ukraina",
      pt: "Ucrânia",
      nl: "Oekraïne",
      hr: "Ukrajina",
      fa: "وکراین",
      de: "Ukraine",
      es: "Ucrania",
      fr: "Ukraine",
      ja: "ウクライナ",
      it: "Ucraina",
      hu: "Ukrajna",
    },
    flag: "https://flagcdn.com/ua.svg",
    cioc: "UKR",
    independent: true,
  },
];

export const serverResponseHungary = [
  {
    name: "Hungary",
    topLevelDomain: [".hu"],
    alpha2Code: "HU",
    alpha3Code: "HUN",
    callingCodes: ["36"],
    capital: "Budapest",
    altSpellings: ["HU"],
    subregion: "Central Europe",
    region: "Europe",
    population: 9749763,
    latlng: [47.0, 20.0],
    demonym: "Hungarian",
    area: 93028.0,
    gini: 29.6,
    timezones: ["UTC+01:00"],
    borders: ["AUT", "HRV", "ROU", "SRB", "SVK", "SVN", "UKR"],
    nativeName: "Magyarország",
    numericCode: "348",
    flags: {
      svg: "https://flagcdn.com/hu.svg",
      png: "https://flagcdn.com/w320/hu.png",
    },
    currencies: [
      {
        code: "HUF",
        name: "Hungarian forint",
        symbol: "Ft",
      },
    ],
    languages: [
      {
        iso639_1: "hu",
        iso639_2: "hun",
        name: "Hungarian",
        nativeName: "magyar",
      },
    ],
    translations: {
      br: "Hungaria",
      pt: "Hungria",
      nl: "Hongarije",
      hr: "Mađarska",
      fa: "مجارستان",
      de: "Ungarn",
      es: "Hungría",
      fr: "Hongrie",
      ja: "ハンガリー",
      it: "Ungheria",
      hu: "Magyarország",
    },
    flag: "https://flagcdn.com/hu.svg",
    regionalBlocs: [
      {
        acronym: "EU",
        name: "European Union",
      },
    ],
    cioc: "HUN",
    independent: true,
  },
];

export const serverResponseCanada = [
  {
    name: "Canada",
    topLevelDomain: [".ca"],
    alpha2Code: "CA",
    alpha3Code: "CAN",
    callingCodes: ["1"],
    capital: "Ottawa",
    altSpellings: ["CA"],
    subregion: "Northern America",
    region: "Americas",
    population: 38005238,
    latlng: [60.0, -95.0],
    demonym: "Canadian",
    area: 9984670.0,
    gini: 33.3,
    timezones: [
      "UTC-08:00",
      "UTC-07:00",
      "UTC-06:00",
      "UTC-05:00",
      "UTC-04:00",
      "UTC-03:30",
    ],
    borders: ["USA"],
    nativeName: "Canada",
    numericCode: "124",
    flags: {
      svg: "https://flagcdn.com/ca.svg",
      png: "https://flagcdn.com/w320/ca.png",
    },
    currencies: [
      {
        code: "CAD",
        name: "Canadian dollar",
        symbol: "$",
      },
    ],
    languages: [
      {
        iso639_1: "en",
        iso639_2: "eng",
        name: "English",
        nativeName: "English",
      },
      {
        iso639_1: "fr",
        iso639_2: "fra",
        name: "French",
        nativeName: "français",
      },
    ],
    translations: {
      br: "Kanada",
      pt: "Canadá",
      nl: "Canada",
      hr: "Kanada",
      fa: "کانادا",
      de: "Kanada",
      es: "Canadá",
      fr: "Canada",
      ja: "カナダ",
      it: "Canada",
      hu: "Kanada",
    },
    flag: "https://flagcdn.com/ca.svg",
    regionalBlocs: [
      {
        acronym: "NAFTA",
        name: "North American Free Trade Agreement",
        otherNames: [
          "Tratado de Libre Comercio de América del Norte",
          "Accord de Libre-échange Nord-Américain",
        ],
      },
    ],
    cioc: "CAN",
    independent: true,
  },
];

export const serverResponseRussia = [
  {
    name: "Belarus",
    topLevelDomain: [".by"],
    alpha2Code: "BY",
    alpha3Code: "BLR",
    callingCodes: ["375"],
    capital: "Minsk",
    altSpellings: [
      "BY",
      "Bielaruś",
      "Republic of Belarus",
      "Белоруссия",
      "Республика Беларусь",
      "Belorussiya",
      "Respublika Belarus’",
    ],
    subregion: "Eastern Europe",
    region: "Europe",
    population: 9398861,
    latlng: [53.0, 28.0],
    demonym: "Belarusian",
    area: 207600.0,
    gini: 25.3,
    timezones: ["UTC+03:00"],
    borders: ["LVA", "LTU", "POL", "RUS", "UKR"],
    nativeName: "Белару́сь",
    numericCode: "112",
    flags: {
      svg: "https://flagcdn.com/by.svg",
      png: "https://flagcdn.com/w320/by.png",
    },
    currencies: [
      {
        code: "BYN",
        name: "New Belarusian ruble",
        symbol: "Br",
      },
      {
        code: "BYR",
        name: "Old Belarusian ruble",
        symbol: "Br",
      },
    ],
    languages: [
      {
        iso639_1: "be",
        iso639_2: "bel",
        name: "Belarusian",
        nativeName: "беларуская мова",
      },
      {
        iso639_1: "ru",
        iso639_2: "rus",
        name: "Russian",
        nativeName: "Русский",
      },
    ],
    translations: {
      br: "Belarus",
      pt: "Bielorrússia",
      nl: "Wit-Rusland",
      hr: "Bjelorusija",
      fa: "بلاروس",
      de: "Weißrussland",
      es: "Bielorrusia",
      fr: "Biélorussie",
      ja: "ベラルーシ",
      it: "Bielorussia",
      hu: "Fehéroroszország",
    },
    flag: "https://flagcdn.com/by.svg",
    regionalBlocs: [
      {
        acronym: "EEU",
        name: "Eurasian Economic Union",
        otherAcronyms: ["EAEU"],
      },
    ],
    cioc: "BLR",
    independent: true,
  },
  {
    name: "Brunei Darussalam",
    topLevelDomain: [".bn"],
    alpha2Code: "BN",
    alpha3Code: "BRN",
    callingCodes: ["673"],
    capital: "Bandar Seri Begawan",
    altSpellings: ["BN", "Nation of Brunei", " the Abode of Peace"],
    subregion: "South-Eastern Asia",
    region: "Asia",
    population: 437483,
    latlng: [4.5, 114.66666666],
    demonym: "Bruneian",
    area: 5765.0,
    timezones: ["UTC+08:00"],
    borders: ["MYS"],
    nativeName: "Negara Brunei Darussalam",
    numericCode: "096",
    flags: {
      svg: "https://flagcdn.com/bn.svg",
      png: "https://flagcdn.com/w320/bn.png",
    },
    currencies: [
      {
        code: "BND",
        name: "Brunei dollar",
        symbol: "$",
      },
      {
        code: "SGD",
        name: "Singapore dollar",
        symbol: "$",
      },
    ],
    languages: [
      {
        iso639_1: "ms",
        iso639_2: "msa",
        name: "Malay",
        nativeName: "bahasa Melayu",
      },
    ],
    translations: {
      br: "Brunei",
      pt: "Brunei",
      nl: "Brunei",
      hr: "Brunej",
      fa: "برونئی",
      de: "Brunei",
      es: "Brunei",
      fr: "Brunei",
      ja: "ブルネイ・ダルサラーム",
      it: "Brunei",
      hu: "Brunei",
    },
    flag: "https://flagcdn.com/bn.svg",
    regionalBlocs: [
      {
        acronym: "ASEAN",
        name: "Association of Southeast Asian Nations",
      },
    ],
    cioc: "BRU",
    independent: true,
  },
  {
    name: "Cyprus",
    topLevelDomain: [".cy"],
    alpha2Code: "CY",
    alpha3Code: "CYP",
    callingCodes: ["357"],
    capital: "Nicosia",
    altSpellings: [
      "CY",
      "Kýpros",
      "Kıbrıs",
      "Republic of Cyprus",
      "Κυπριακή Δημοκρατία",
      "Kıbrıs Cumhuriyeti",
    ],
    subregion: "Southern Europe",
    region: "Europe",
    population: 1207361,
    latlng: [35.0, 33.0],
    demonym: "Cypriot",
    area: 9251.0,
    gini: 32.7,
    timezones: ["UTC+02:00"],
    borders: ["GBR"],
    nativeName: "Κύπρος",
    numericCode: "196",
    flags: {
      svg: "https://flagcdn.com/cy.svg",
      png: "https://flagcdn.com/w320/cy.png",
    },
    currencies: [
      {
        code: "EUR",
        name: "Euro",
        symbol: "€",
      },
    ],
    languages: [
      {
        iso639_1: "el",
        iso639_2: "ell",
        name: "Greek (modern)",
        nativeName: "ελληνικά",
      },
      {
        iso639_1: "tr",
        iso639_2: "tur",
        name: "Turkish",
        nativeName: "Türkçe",
      },
      {
        iso639_1: "hy",
        iso639_2: "hye",
        name: "Armenian",
        nativeName: "Հայերեն",
      },
    ],
    translations: {
      br: "Kiprenez",
      pt: "Chipre",
      nl: "Cyprus",
      hr: "Cipar",
      fa: "قبرس",
      de: "Zypern",
      es: "Chipre",
      fr: "Chypre",
      ja: "キプロス",
      it: "Cipro",
      hu: "Ciprus",
    },
    flag: "https://flagcdn.com/cy.svg",
    regionalBlocs: [
      {
        acronym: "EU",
        name: "European Union",
      },
    ],
    cioc: "CYP",
    independent: true,
  },
  {
    name: "Russian Federation",
    topLevelDomain: [".ru"],
    alpha2Code: "RU",
    alpha3Code: "RUS",
    callingCodes: ["7"],
    capital: "Moscow",
    altSpellings: [
      "RU",
      "Rossiya",
      "Russian Federation",
      "Российская Федерация",
      "Rossiyskaya Federatsiya",
    ],
    subregion: "Eastern Europe",
    region: "Europe",
    population: 144104080,
    latlng: [60.0, 100.0],
    demonym: "Russian",
    area: 17124442,
    gini: 37.5,
    timezones: [
      "UTC+03:00",
      "UTC+04:00",
      "UTC+06:00",
      "UTC+07:00",
      "UTC+08:00",
      "UTC+09:00",
      "UTC+10:00",
      "UTC+11:00",
      "UTC+12:00",
    ],
    borders: [
      "AZE",
      "BLR",
      "CHN",
      "EST",
      "FIN",
      "GEO",
      "KAZ",
      "PRK",
      "LVA",
      "LTU",
      "MNG",
      "NOR",
      "POL",
      "UKR",
    ],
    nativeName: "Россия",
    numericCode: "643",
    flags: {
      svg: "https://flagcdn.com/ru.svg",
      png: "https://flagcdn.com/w320/ru.png",
    },
    currencies: [
      {
        code: "RUB",
        name: "Russian ruble",
        symbol: "₽",
      },
    ],
    languages: [
      {
        iso639_1: "ru",
        iso639_2: "rus",
        name: "Russian",
        nativeName: "Русский",
      },
    ],
    translations: {
      br: "Rusia",
      pt: "Rússia",
      nl: "Rusland",
      hr: "Rusija",
      fa: "روسیه",
      de: "Russland",
      es: "Rusia",
      fr: "Russie",
      ja: "ロシア連邦",
      it: "Russia",
      hu: "Oroszország",
    },
    flag: "https://flagcdn.com/ru.svg",
    regionalBlocs: [
      {
        acronym: "EEU",
        name: "Eurasian Economic Union",
        otherAcronyms: ["EAEU"],
      },
    ],
    cioc: "RUS",
    independent: true,
  },
];

export const serverResponseCountryDetails = {
  Ukraine: serverResponseUkraine,
  Canada: serverResponseCanada,
  Hungary: serverResponseHungary,
  Russia: serverResponseRussia,
  "United%20States%20of%20America": serverResponseUSA,
};

export const serverResponseAlphaCodes = {
  RUS: serverResponseRussia,
  USA: serverResponseUSA,
  CAN: serverResponseCanada,
  MEX: serverResponseMexico,
  UKR: serverResponseUkraine,
  HUN: serverResponseHungary,
};
