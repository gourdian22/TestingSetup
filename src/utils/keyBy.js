export const keyBy = (collection, grouper) => {
  const result = {}; // создаем пустой объект
  const getKey = (elem) =>
    typeof grouper === "function" ? grouper(elem) : elem[grouper]; // создаем функцию которая будет получать ключ, если передали колбек то ключем будет результат колбека на каждом элементе
  const itarateCollaction = Array.isArray(collection)
    ? collection
    : Object.values(collection); // проверяем что нам передали, если объект то нужно взять все его значения (так работает оригинал в lodash)

  itarateCollaction.forEach((value) => {
    // итерируемся по переданной коллекции
    const key = getKey(value); // получаем ключ
    result[key] = value; // тупо перезаписываем старое значение если оно было, если нет пишем новое
  });

  return result;
};
