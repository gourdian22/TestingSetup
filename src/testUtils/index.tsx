/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable react/require-default-props */
import {
  render as rtlRender,
  RenderOptions,
  RenderResult,
} from "@testing-library/react";

import { ReactElement, ReactNode } from "react";
import { Provider } from "react-redux";
import {
  RouteObject,
  RouterProvider,
  createMemoryRouter,
} from "react-router-dom";
import { router } from "../App";

import { configStoreWithInitialState, RootState } from "store";

type CustomRenderOptions = {
  preloadedState?: Partial<RootState>;
  routeHistory?: Array<string>;
  initialRouteIndex?: number; // index in the routeHistory array to start the test
  renderOptions?: Omit<RenderOptions, "wrapper">;
  routes?: RouteObject[];
};

type CustomRouter = typeof router;

type CustomRenderResult = RenderResult & { router: CustomRouter };

type MemoryHistoryBuildOptions = {
  initialEntries?: string[];
  initialIndex?: number;
};
function render(
  ui: ReactElement,
  {
    preloadedState = {},
    routeHistory = [],
    initialRouteIndex,
    routes,
    ...renderOptions
  }: CustomRenderOptions = {}
): CustomRenderResult {
  const memoryHistoryArgs: MemoryHistoryBuildOptions = {};
  if (routeHistory.length > 0) {
    memoryHistoryArgs.initialEntries = routeHistory;
    memoryHistoryArgs.initialIndex = initialRouteIndex;
  }

  let customRouter = createMemoryRouter(router.routes, {
    ...memoryHistoryArgs,
  });
  if (routes) {
    customRouter = createMemoryRouter(routes, { ...memoryHistoryArgs });
  }
  function Wrapper(): ReactElement {
    const store = configStoreWithInitialState(preloadedState);

    return (
      <Provider store={store}>
        <RouterProvider router={customRouter}></RouterProvider>
      </Provider>
    );
  }

  const renderResult = rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
  return { ...renderResult, router: customRouter };
}

// re-export everything
export * from "@testing-library/react";

// override render method and export history
export { render };
