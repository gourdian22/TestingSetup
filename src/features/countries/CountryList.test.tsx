import { render, screen, waitFor } from "../../testUtils";
import { CountryList } from "./CountryList";
import userEvent from "@testing-library/user-event";
import { App } from "../../App";
import { keyBy } from "../../utils/keyBy";
import {
  serverResponseAllCountries,
  serverResponseCountryDetails,
  serverResponseAlphaCodes,
} from "../../testUtils/fakeData";

it("should render list of countries", async () => {
  render(<CountryList />);
  const serverCountriesNames = keyBy(serverResponseAllCountries, "name");
  const countriesArticles = await screen.findAllByRole("article");
  expect(countriesArticles).toHaveLength(3);
  expect(countriesArticles).toMatchSnapshot();
  await waitFor(async () => {
    const countriesNames = await screen.findAllByRole("heading");
    countriesNames.forEach((element) => {
      expect(element.textContent).toBe(
        //@ts-ignore
        serverCountriesNames[element.textContent].name
      );
    });
  });
});

it("should show country details page after click", async () => {
  const user = userEvent.setup();
  const { router } = render(<App />);
  const canadaImage = await screen.findByRole("img", {
    name: /Canada/i,
  });
  await user.click(canadaImage);
  expect(router.state.location.pathname).toBe("/country/Canada");
});

it("should show country details with borders", async () => {
  const user = userEvent.setup();
  render(<CountryList />);
  const canadaImage = await screen.findByRole("img", {
    name: /Canada/i,
  });
  await user.click(canadaImage);
  const neighbors = await screen.findAllByTestId("meta");
  const neighborsNames = neighbors.map((neighbor) => neighbor.textContent);

  const serverCanadaNeigborsAlphaCodes =
    serverResponseCountryDetails["Canada"][0]["borders"];

  const serverCanadaNeighborsNames = serverCanadaNeigborsAlphaCodes.map(
    (code) => {
      return serverResponseAlphaCodes[
        code as keyof typeof serverResponseAlphaCodes
      ][0].name;
    }
  );
  expect(neighborsNames).toEqual(serverCanadaNeighborsNames);
});
