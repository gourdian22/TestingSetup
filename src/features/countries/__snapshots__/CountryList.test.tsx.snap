// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render list of countries 1`] = `
Array [
  <article
    class="sc-kDTinF gQCeoh"
  >
    <img
      alt="Canada"
      class="sc-iqseJM bliWJi"
      src="https://flagcdn.com/w320/ca.png"
    />
    <div
      class="sc-crHmcD beotCT"
    >
      <h3
        class="sc-egiyK bMagzs"
      >
        Canada
      </h3>
      <ul
        class="sc-bqiRlB cPTZjy"
      >
        <li
          class="sc-ksdxgE cKINCt"
        >
          <b>
            Population
            :
          </b>
           
          38 005 238
        </li>
        <li
          class="sc-ksdxgE cKINCt"
        >
          <b>
            Region
            :
          </b>
           
          Americas
        </li>
        <li
          class="sc-ksdxgE cKINCt"
        >
          <b>
            Capital
            :
          </b>
           
          Ottawa
        </li>
      </ul>
    </div>
  </article>,
  <article
    class="sc-kDTinF gQCeoh"
  >
    <img
      alt="Mexico"
      class="sc-iqseJM bliWJi"
      src="https://flagcdn.com/w320/mx.png"
    />
    <div
      class="sc-crHmcD beotCT"
    >
      <h3
        class="sc-egiyK bMagzs"
      >
        Mexico
      </h3>
      <ul
        class="sc-bqiRlB cPTZjy"
      >
        <li
          class="sc-ksdxgE cKINCt"
        >
          <b>
            Population
            :
          </b>
           
          128 932 753
        </li>
        <li
          class="sc-ksdxgE cKINCt"
        >
          <b>
            Region
            :
          </b>
           
          Americas
        </li>
        <li
          class="sc-ksdxgE cKINCt"
        >
          <b>
            Capital
            :
          </b>
           
          Mexico City
        </li>
      </ul>
    </div>
  </article>,
  <article
    class="sc-kDTinF gQCeoh"
  >
    <img
      alt="United States of America"
      class="sc-iqseJM bliWJi"
      src="https://flagcdn.com/w320/us.png"
    />
    <div
      class="sc-crHmcD beotCT"
    >
      <h3
        class="sc-egiyK bMagzs"
      >
        United States of America
      </h3>
      <ul
        class="sc-bqiRlB cPTZjy"
      >
        <li
          class="sc-ksdxgE cKINCt"
        >
          <b>
            Population
            :
          </b>
           
          329 484 123
        </li>
        <li
          class="sc-ksdxgE cKINCt"
        >
          <b>
            Region
            :
          </b>
           
          Americas
        </li>
        <li
          class="sc-ksdxgE cKINCt"
        >
          <b>
            Capital
            :
          </b>
           
          Washington, D.C.
        </li>
      </ul>
    </div>
  </article>,
]
`;
